plugins/admin_console/services/EntryAdminService.php

```diff
@@ -81,4 +93,113 @@ function getTracksAction($entryId)
 		$response->totalCount = count($dbList);
 		return $response;
 	}
+
+	/**
+	 * Restore deleted entry.
+	 *
+	 * @action restoreDeletedEntry
+	 * @param string $entryId
+	 * @return KalturaBaseEntry The restored entry
+	 */
+	public function restoreDeletedEntryAction($entryId)
+	{
+		$deletedEntry = entryPeer::retrieveByPKNoFilter($entryId);
+		if (!$deletedEntry)
+			throw new KalturaAPIException(KalturaErrors::ENTRY_ID_NOT_FOUND, $entryId);
+
+		$fileSyncKeys = array();
+		foreach (self::$fileSyncKeysToRestore as $key) {
+			$fileSyncKeys[] = $deletedEntry->getSyncKey($key);
+		}
+
+		$c = new Criteria();
+		$c->add(assetPeer::ENTRY_ID, $entryId, Criteria::EQUAL);
+		assetPeer::setUseCriteriaFilter(false);
+		$deletedAssets = assetPeer::doSelect($c);
+		assetPeer::setUseCriteriaFilter(true);
+
+		foreach($deletedAssets as $deletedAsset)
+		{
+			array_push($fileSyncKeys, $deletedAsset->getSyncKey(asset::FILE_SYNC_ASSET_SUB_TYPE_ASSET), $deletedAsset->getSyncKey(asset::FILE_SYNC_ASSET_SUB_TYPE_CONVERT_LOG));
+		}
+
+		$fileSyncs = array();
+		FileSyncPeer::setUseCriteriaFilter(false);
+		foreach ($fileSyncKeys as $fileSyncKey)
+		{
+			$fileSyncs = array_merge($fileSyncs, FileSyncPeer::retrieveAllByFileSyncKey($fileSyncKey));
+		}
+		FileSyncPeer::setUseCriteriaFilter(true);
+
+		if (!$this->validateEntryForRestoreDelete($deletedEntry, $fileSyncs, $deletedAssets))
```

Comment:
I think you better throw the exception inside validateEntryForRestoreDelete in case you would like to validate more conditions in the future.